### PR TITLE
Populate metering report json [DPP-1107]

### PR DESCRIPTION
**Pull request by simonmaxen-da**

### CODE REVIEW

**Comment by simonmaxen-da** [comment]
Sandbox output looks as expected:
```
grpcurl --plaintext -d '{"from":"2020-01-01T00:00:00Z", "application_id": "AppX"}' localhost:6865 com.daml.ledger.api.v1.admin.MeteringReportService.GetMeteringReport
{
  "request": {
    "from": "2020-01-01T00:00:00Z",
    "application_id": "AppX"
  },
  "participant_report": {
    "participant_id": "default"
  },
  "report_generation_time": "2022-07-28T16:15:06.221660Z",
  "metering_report_json": {
      "applications": [
          ],
      "final": false,
      "participant": "default",
      "request": {
            "application": "AppX",
            "from": "2020-01-01T00:00:00Z"
          }
    }
}
```

**Comment by simonmaxen-da** [comment]
Previous sandbox output:
```
grpcurl --plaintext -d '{"from":"2020-01-01T00:00:00Z", "application_id": "AppX"}' localhost:6865 com.daml.ledger.api.v1.admin.MeteringReportService.GetMeteringReport | jq
{
  "request": {
    "from": "2020-01-01T00:00:00Z",
    "application_id": "AppX"
  },
  "participant_report": {
    "participant_id": "default"
  },
  "report_generation_time": "2022-07-28T16:24:58.470954Z"
}
```

File: ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/meteringreport/MeteringReportSpec.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+class MeteringReportSpec extends AsyncWordSpec with Matchers {
+
+  "the metering report" should {
+
+    "serialize and deserialize as JSON" in {
+
+      val application = Ref.ApplicationId.assertFromString("a0")
+      val from = Timestamp.now()
+      val to = from.add(Duration.of(-1, ChronoUnit.DAYS))
```
Comment (mziolekda) [inline ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/meteringreport/MeteringReportSpec.scala:24]: I don't get it, is `to` 1 day before `from`?

**Review by mziolekda — APPROVED** [review]

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiMeteringReportService.scala [hunk 2]
```diff
@@ -99,26 +103,54 @@ private[apiserver] object ApiMeteringReportService {
...
+      GetMeteringReportResponse(
+        request = Some(request),
+        participantReport = Some(genParticipantReport(reportData)),
+        reportGenerationTime = Some(generationTime),
+        meteringReportJson = Some(genMeteringReportJson(from, to, applicationId, reportData)),
+      )
+
+    }
+
+    // Note that this will be removed once downstream consumers no longer need it
```
Comment (pbatko-da) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiMeteringReportService.scala:122]: I'd have thought that Json report is the new standard, and the non-Json one will get eventually removed

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiMeteringReportService.scala [hunk 3]
```diff
@@ -99,26 +103,54 @@ private[apiserver] object ApiMeteringReportService {
...
+    private def genMeteringReportJson(
+        from: Timestamp,
+        to: Option[Timestamp],
+        applicationId: Option[ApplicationId],
+        reportData: ReportData,
+    ) = {
+
```
Comment (pbatko-da) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiMeteringReportService.scala:129]: Nit: redundant blank line

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiMeteringReportService.scala [hunk 4]
```diff
@@ -99,26 +103,54 @@ private[apiserver] object ApiMeteringReportService {
...
       val applicationReports = reportData.applicationData.toList
+        .sortBy(_._1)
+        .map((ApplicationReport.apply _).tupled)
+
```
Comment (pbatko-da) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiMeteringReportService.scala:133]: Nit: redundant blank line

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiMeteringReportService.scala [hunk 5]
```diff
@@ -99,26 +103,54 @@ private[apiserver] object ApiMeteringReportService {
...
+      val report: ParticipantReport = ParticipantReport(
+        participant = participantId,
+        request = Request(from, to, applicationId),
+        `final` = reportData.isFinal,
+        applications = applicationReports,
+      )
+
```
Comment (pbatko-da) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiMeteringReportService.scala:140]: Nit: redundant blank line

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiMeteringReportService.scala [hunk 6]
```diff
@@ -99,26 +103,54 @@ private[apiserver] object ApiMeteringReportService {
...
+    }
+
+    // Note that this will be removed once downstream consumers no longer need it
+    private def genMeteringReportJson(
+        from: Timestamp,
+        to: Option[Timestamp],
+        applicationId: Option[ApplicationId],
+        reportData: ReportData,
+    ) = {
```
Comment (pbatko-da) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiMeteringReportService.scala:128]: What's the return type?

File: ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/meteringreport/MeteringReportSpec.scala [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+      val expected = ParticipantReport(
+        participant = Ref.ParticipantId.assertFromString("p0"),
+        request = Request(from, Some(to), Some(application)),
+        `final` = false,
+        applications = Seq(ApplicationReport(application, 272)),
+      )
+
+      val json = expected.toJson.prettyPrint
+      // println(s"json: $json")
```
Comment (pbatko-da) [inline ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/meteringreport/MeteringReportSpec.scala:34]: Commented code

File: ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/meteringreport/MeteringReportSpec.scala (hunk 1 above)
Comment (pbatko-da) [inline ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/meteringreport/MeteringReportSpec.scala:24]: Why `-1`? Seems odd

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiMeteringReportService.scala (hunk 2 above)
Comment (simonmaxen-da) [reply ledger/participant-integration-api/src/main/scala/platform/apiserver/services/admin/ApiMeteringReportService.scala:122]: There still needs to be a `GetMeteringReportResponse` the `participantReport` will be removed the new `meteringReportJson` will live on.

File: ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/meteringreport/MeteringReportSpec.scala (hunk 7 above)
Comment (simonmaxen-da) [reply ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/meteringreport/MeteringReportSpec.scala:34]: Removed

File: ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/meteringreport/MeteringReportSpec.scala (hunk 1 above)
Comment (simonmaxen-da) [reply ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/meteringreport/MeteringReportSpec.scala:24]: Made +1

Comment (simonmaxen-da) [reply ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/meteringreport/MeteringReportSpec.scala:24]: Changed to +1

**Pull request merged**